src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -338,7 +339,18 @@ protected void cleanPayeeACHAccountExtractDetail(PayeeACHAccountExtractDetail de
             if (!StringUtils.isNumeric(detail.getBankAccountNumber())) {
                 LOG.error(logMessageStarter + " is not numeric after cleaning");
             }
+
+            detail.setBankName(extractBankName(detail));
+        }
+    }
+
+    private String extractBankName(PayeeACHAccountExtractDetail detail) {
+        String bankName = ObjectUtils.isNull(detail) ? StringUtils.EMPTY : detail.getBankName();
+        if (StringUtils.length(bankName) > 40) {
+            LOG.info("cleanPayeeACHAccountExtractDetail truncating bank name to 40 characters");
+            bankName = bankName.substring(0, 40);
         }
+        return bankName.trim();

```

Comment:
Should the bank name also be trimmed prior to performing the length check? Or do you only need it here?

Also, I think StringUtils has a trim() method on it, if you want to make this null-safe.

Reply:
Ah, nice! A null-safe class method. That's an elegant solution to an unrelated issue, in a different language, that I'm having. I'm glad you suggested it here, and I'm glad I read this.